sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReceiveMessageAndSettleSample.java

```diff
@@ -0,0 +1,100 @@
+        // "<<queue-name>>" will be the name of the Service Bus queue instance you created
+        // inside the Service Bus namespace.
+
+        ServiceBusReceiverAsyncClient receiverAsyncClient = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .buildReceiverClientBuilder()
+            .receiveMode(ReceiveMode.PEEK_LOCK)
+            .isLockAutoRenewed(true)
+            .queueName("<<queue-name>>")
+            .maxAutoLockRenewalDuration(Duration.ofSeconds(2))
+            .buildAsyncClient();
+
+        Disposable subscription = receiverAsyncClient.receive()
+            .subscribe(received -> {
+                Instant latest = Instant.MIN;
+
+                // Simulate some sort of long processing. Sleep should not be used in production system.
+                for (int i = 0; i < 3; i++) {
+                    try {
+                        TimeUnit.SECONDS.sleep(15);
+                    } catch (InterruptedException error) {
+                        System.out.println("Error occurred while sleeping: " + error);
+                    }
+                    latest = received.getLockedUntil();
+                    System.out.println("Message Locked Until " + latest);
+                }
+
+                //This is application business logic to take action based on some application logic.
+                // For demo purpose we are using a property for application logic.
+                String propertyValue = "UNDEFINED";
+                if (received.getProperties().get("SOME_USER_PROPERTY") != null) {
+                    propertyValue = (String) received.getProperties().get("SOME_USER_PROPERTY");
+                }
+
+                switch (propertyValue) {
+                    case "VALUE-1":
+                        System.out.println("Completing message.");
+                        receiverAsyncClient.complete(received).block(Duration.ofSeconds(15));
+                        break;
+                    case "VALUE-2":
+                        System.out.println("Abandon message.");
+                        receiverAsyncClient.abandon(received).block(Duration.ofSeconds(15));
+                        break;
+                    case "VALUE-3":
+                        System.out.println("Defer message.");
+                        receiverAsyncClient.defer(received).block(Duration.ofSeconds(15));
+                        break;
+                    default:
+                        System.out.println("Deadletter message.");
+                        receiverAsyncClient.deadLetter(received).block(Duration.ofSeconds(15));
+                }
+            });
+
+        // Receiving messages from the queue for a duration of 20 seconds.
+        // Subscribe is not a blocking call so we sleep here so the program does not end.
+        try {
+            Thread.sleep(Duration.ofSeconds(20).toMillis());
```

Comment:
You sleep for 20 seconds, but each iteration is 15 seconds in the subscribe. Have you run these samples?